File: dataset_reduzido/avatar/Python/Code/codeforces_299_A.py
num_inp = lambda : int ( input ( ) )
arr_inp = lambda : list ( map ( int , input ( ).split ( ) ) )
sp_inp = lambda : map ( int , input ( ).split ( ) )
a = sorted ( map ( int , [ * open ( 0 ) ] [ 1 ].split ( ) ) )
print ( [ a [ 0 ] , - 1 ] [ any ( x % a [ 0 ] for x in a ) ] )

File: output/Llama-2-100/avatar/Python/Java/codeforces_299_A.java
public class codeforces_299_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int num_inp = sc.nextInt();
        List<Integer> arr_inp = Arrays.asList(sc.nextLine().split(" "));
        List<Integer> sp_inp = Arrays.asList(sc.nextLine().split(" "));
        List<Integer> a = Arrays.asList(sc.nextLine().split(" "));
        int result = -1;
        for (int x : a) {
            if (x % num_inp == 0) {
                result = x;
                break;
            }
        }
        System.out.println(result);
    }
}


